pkg/cmd/kafka/create/create.go

```diff
@@ -189,6 +190,10 @@ func runCreate(opts *options) error {
 		return opts.localizer.MustLocalizeError("kafka.create.error.conflictError", localize.NewEntry("Name", payload.Name))
 	}
 
+	if kas.IsErr(err, kas.OneEvalInstanceAllowed) {
```

Comment:
Getting erroneously evaluated as false.
![Screenshot from 2021-09-24 12-19-56](https://user-images.githubusercontent.com/23582438/134630936-76ee744f-e9ba-4cb9-ab0c-73cafb0ca380.png)

Reply:
I looked into this and it appears that the error dooes not properly encode the metadata within the error, it is stringified:

![image](https://user-images.githubusercontent.com/11743717/134645454-3c318ed8-96b1-4b5e-9611-250a7463cf9f.png)

I checked another error on the same endpoint but it acts as expected:

![image](https://user-images.githubusercontent.com/11743717/134644540-944da43d-e7fb-4d78-af9a-ea677f4039a8.png)

Reply:
Sorry for merge without review.
Feel free to reopen/revert this change

Reply:
Hmm but this is not merged?

Reply:
Got notification from other PR and got confused. @craicoverflow This looks like sdk bug related to using very unusual http status code on backend. I would recomend us to change status code in sdk to confirm that this is the case then ping backend to correct status code to 400 or something

Reply:
429 is ratelimiting status code and it never carries payload etc. @machi1990

Reply:
> 429 is ratelimiting status code and it never carries payload etc. @machi1990

Are you not mixing this status code with `204`? 

> Seems like not valid use for this error

What error would you expect for this case? Does the ui face the same problem? I think they integrated it already. 

pinging @ziccardi for additional input

Reply:
429 is specifically for rate-limiting, which is not what is happening in this scenario.

Reply:
Same problem happens for UI and this code is not handled  in UI right now:
https://github.com/bf2fc6cc711aee1a0c2a/kas-ui/blob/1c98ca3820e7574529e8251a6c146a49bf6484a8/src/app/modules/OpenshiftStreams/dialogs/CreateInstance/CreateInstance.tsx#L259
ErrorCodes.INSUFFICIENT_QUOTA is still used. Not sure if that code is returned from backend/not dead?

To avoid team introducing new codes and then passing them to indivudual cli,ui devs we need to come up with strategy for this.

What we should do now
- Correct error code on backend to 400 (control plane team) @machi1990 to confirm
- Adjust UI and CLI to new code once it is there (devexp)
- Raise topic on architecture call to figure out how to handle this problem in future. (devexp done some spikes for this to avoid issues like this one) (Devexp)
-

Reply:
To confirm:

- No one uses this error right now
- we cannot make clients to use this type of error as we will be introducing security issue

Reply:
When we implemented the feature in the control plane, we had a discussion about the return code to be used and in the end we opted for 429. In our case, it can be seen as 'exceeding the rate of 1 instance every 48 hours' (after 48 hours the eval instance is deleted). 

According to [RFC-6585](https://datatracker.ietf.org/doc/html/rfc6585#section-4) 429 errors should carry the details of what has happened.

Reply:
Lets hear our arguments on this. Can we can have short call next week to discuss this?

Reply:
As a workaround until then, I have added a check based on the status code: 0ff83ee76c3b90381b551aedcaf28d8365ca73e1